### PR TITLE
`azurerm_redis_cache_access_policy` - remove `ForceNew` from `permissions`

### LINKED ISSUE #26390: Access policy is replaced when permissions are changed
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.8.5

### AzureRM Provider Version

3.108

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache_access_policy

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "=3.108.0"
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "East US"
}

resource "azurerm_redis_cache" "trf-example" {
  name                = "trf-example"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  capacity            = 1
  family              = "P"
  sku_name            = "Premium"
  enable_non_ssl_port = false

  redis_configuration {
    maxmemory_reserved = 2
    maxmemory_delta    = 2
    maxmemory_policy   = "allkeys-lru"
  }
}

resource "azurerm_redis_cache_access_policy" "example" {
  name           = "example"
  redis_cache_id = azurerm_redis_cache.trf-example.id
  permissions    = "+@read +@connection +cluster|info"
}

resource "azurerm_redis_cache_access_policy_assignment" "example" {
  name                = "00000000-0000-0000-0000-000000000000"
  redis_cache_id      = azurerm_redis_cache.trf-example.id
  access_policy_name  = azurerm_redis_cache_access_policy.example.name
  object_id           = "00000000-0000-0000-0000-000000000000"
  object_id_alias     = "ServicePrincipal"
}
```

### Debug Output/Panic Output

```shell
│ Error: deleting Redis Cache Access Policy example in Redis Cache trf-example in resource group example-resources: performing AccessPolicyDelete: unexpected status 400 (400 Bad Request) with error: AccessPolicyUpdateNotPermitted: Cannot delete access policy while it is linked to access policy assignment. For more information, see: https://aka.ms/redis/MicrosoftEntraAuthenticationPrerequisites.
│
│ deleting Redis Cache Access Policy example in Redis Cache trf-example in resource group example-resources: performing
│ AccessPolicyDelete: unexpected status 400 (400 Bad Request) with error: AccessPolicyUpdateNotPermitted: Cannot delete access policy
│ while it is linked to access policy assignment. For more information, see:
│ https://aka.ms/redis/MicrosoftEntraAuthenticationPrerequisites.
```

### Expected Behaviour

When the permissions of an access policy are changed, the access policy should update itself. It should not delete and recreate itself.

### Actual Behaviour

An access policy that has been assigned to an object id cannot be deleted. When the permissions of an access policy are updated, terraform tries to delete and recreate the access policy, which fails since the access policy is assigned to an object id.

### Steps to Reproduce

1. `terraform apply` the configuration above
2. Change resource azurerm_redis_cache_access_policy.example.name.permissions to `+@read +@write +@connection +cluster|info`
3. `terraform apply` again
4. This should fail with the error shared above

### Important Factoids

_No response_

### References

Issue that was raised for introducing support for MS Entra Authentication https://github.com/hashicorp/terraform-provider-azurerm/issues/23371

### PR DESCRIPTION
**Pull request by sinbai**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

After validating the behavior of the API, the `permissions` is allowed to be updated. Submit this PR to remove the `ForceNew` for `permissions` to fix #26390 

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

`PASS: TestAccRedisCacheAccessPolicy_update (2489.80s)
`
## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_redis_cache_access_policy` -  remove `ForceNew` from `permissions` [GH-26390]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #26390

### CODE REVIEW

File: internal/services/redis/redis_cache_access_policy_resource.go [hunk 1]
```diff
@@ -112,6 +110,52 @@ func (r RedisCacheAccessPolicyResource) Create() sdk.ResourceFunc {
...
+				return err
+			}
+
+			existing, err := client.AccessPolicyGet(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
```
Comment (stephybun) [inline internal/services/redis/redis_cache_access_policy_resource.go:135]:
```suggestion
				return fmt.Errorf("retrieving %s: `model` was nil", *id)
```

**Review by stephybun — COMMENTED** [review]
Thanks @sinbai, one minor comment on the error message otherwise this should be good to go!

File: internal/services/redis/redis_cache_access_policy_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/redis/redis_cache_access_policy_resource.go]: Fixed.

**Comment by sinbai** [comment]
> Thanks @sinbai, one minor comment on the error message otherwise this should be good to go!

Hi @stephybun thanks for your feedback. I have updated the code. Could you please take another look?

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 👍

**Pull request merged**